core/src/main/java/org/infinispan/persistence/sifs/LogAppender.java

```diff
@@ -233,14 +235,16 @@ public void accept(WriteOperation writeOperation) {
          if (currentOffset != 0 && currentOffset + actualLength > maxFileSize) {
             // switch to next file
             logFile.close();
-            compactor.completeFile(logFile.fileId, currentOffset);
+            compactor.completeFile(logFile.fileId, currentOffset, nextExpirationTime);
```

Comment:
Don't we also need to set `nextExpirationTime = -1;` when `actualRequest.isClear()`?

Reply:
Yes, I will add that in. Good catch.